src/include/migraphx/op/reshape.hpp

```diff
@@ -232,42 +125,37 @@ struct reshape
             }
         }
 
-        auto s = reshape_dims(inputs.front(), rdims);
-        if(not s.has_value())
-            MIGRAPHX_THROW("Reshape on axis that is not packed.");
-
-        if(s->elements() != inputs.front().elements())
-            MIGRAPHX_THROW("Reshape: Wrong number of elements for reshape: reshape has " +
-                           std::to_string(s->elements()) + " elements whereas the input has " +
-                           std::to_string(inputs.front().elements()));
-
-        assert(s->bytes() == inputs.front().bytes());
-        return *s;
+        return {inputs.front().type(), rdims};
     }
 
     shape compute_shape(std::vector<shape> inputs) const
     {
         check_shapes{inputs, *this, true}.has(1);
-        auto n_neg_dims = std::count(dims.begin(), dims.end(), -1);
-        if(n_neg_dims > 1)
-            MIGRAPHX_THROW("Reshape: Dimensions for reshape can only have one -1 dim");
-        auto s0 = inputs[0];
+        auto s0 = inputs.front();
         if(s0.dynamic())
         {
-            return dyn_compute_shape(s0);
+            return s0;
         }
         else
         {
-            return static_compute_shape(inputs, n_neg_dims);
-        }
+            auto t           = s0.type();
+            return {t, dims};
+        } 
     }
 
     argument compute(const dyn_output& dyn_out, std::vector<argument> args) const
     {
-        return args[0].reshape(dyn_out.computed_shape);
+        assert(dyn_out.computed_shape.standard());
+        argument result{dyn_out.computed_shape};
+
+        visit_all(result, args[0])([&](auto output, auto input) {
+            shape_for_each(output.get_shape(), [&](const auto& idx) {
```

Comment:
Use std::copy.

Reply:
Ah, we planning on speeding this up with std::execution?

Reply:
We might, but either way using `std::copy` is clearer.

Reply:
Done